src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1158,6 +1158,13 @@ func getQuiescentCondition(redpandaCluster *vectorizedv1alpha1.Cluster) vectoriz
 		return condition
 	}
 
+	if redpandaCluster.Status.CurrentReplicas != redpandaCluster.Status.Replicas || redpandaCluster.Status.Replicas != redpandaCluster.Status.ReadyReplicas || (redpandaCluster.Spec.Replicas != nil && *redpandaCluster.Spec.Replicas != redpandaCluster.Status.Replicas) {
```

Comment:
This is a bit of a beast to read and reason about. Could we break this into a couple variables first?
```go
hasCurrentReplicas := redpandaCluster.Status.CurrentReplicas == redpandaCluster.Status.Replicas
// ...
if !(hasCurrentReplicas && hasReadyReplicas /* ... */) {

}
```

Reply:
refactored